### PR TITLE
Map some mostly world gen stuff

### PR DESCRIPTION
**Pull request by valoeghese**

### CODE REVIEW

File: mappings/net/minecraft/class_4719.mapping [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CLASS net/minecraft/class_4719
```
Comment (liach) [inline mappings/net/minecraft/class_4719.mapping]: This is like a sign type, but dunno if mojang will use it for other wood stuff in the future...

File: mappings/net/minecraft/class_4719.mapping [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	FIELD field_21676 OAK Lnet/minecraft/class_4719;
+	FIELD field_21677 SPRUCE Lnet/minecraft/class_4719;
+	FIELD field_21678 BIRCH Lnet/minecraft/class_4719;
+	FIELD field_21679 ACACIA Lnet/minecraft/class_4719;
+	FIELD field_21680 JUNGLE Lnet/minecraft/class_4719;
+	FIELD field_21681 DARK_OAK Lnet/minecraft/class_4719;
+	FIELD field_21683 type Ljava/lang/String;
```
Comment (liach) [inline mappings/net/minecraft/class_4719.mapping]: name
```suggestion
	FIELD field_21683 name Ljava/lang/String;
```

**Comment by valoeghese** [comment]
Done

File: mappings/net/minecraft/world/gen/feature/BranchedTreeFeatureConfig.mapping [hunk 3]
```diff
@@ -1,20 +1,67 @@
...
-	FIELD field_21264 trunkTopOffsetRandom I
+	FIELD field_21264 trunkTopOffset I
+	FIELD field_21265 trunkTopOffsetRandom I
+	FIELD field_21266 foliageHeight I
+	FIELD field_21267 foliageHeightRandom I
+	FIELD field_21268 maxWaterDepth I
+	FIELD field_21269 noVines Z
 	METHOD <init> (Lnet/minecraft/class_4651;Lnet/minecraft/class_4651;Lnet/minecraft/class_4647;Ljava/util/List;IIIIIIIIIIZ)V
 		ARG 1 stateProvider
 		ARG 2 stateProvider2
```
Comment (liach) [inline mappings/net/minecraft/world/gen/feature/BranchedTreeFeatureConfig.mapping]: what is this state provider

**Comment by valoeghese** [comment]
Done
Also named some more tree gen stuff

File: mappings/net/minecraft/util/WoodType.mapping [hunk 4]
```diff
@@ -0,0 +1,14 @@
+CLASS net/minecraft/class_4719 net/minecraft/util/WoodType
+	FIELD field_21676 OAK Lnet/minecraft/class_4719;
+	FIELD field_21677 SPRUCE Lnet/minecraft/class_4719;
+	FIELD field_21678 BIRCH Lnet/minecraft/class_4719;
+	FIELD field_21679 ACACIA Lnet/minecraft/class_4719;
+	FIELD field_21680 JUNGLE Lnet/minecraft/class_4719;
+	FIELD field_21681 DARK_OAK Lnet/minecraft/class_4719;
+	FIELD field_21682 WOODTYPES Ljava/util/Set;
```
Comment (liach) [inline mappings/net/minecraft/util/WoodType.mapping]: `VALUES`?

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/util/WoodType.mapping (hunk 4 above)
Comment (Runemoro) [reply mappings/net/minecraft/util/WoodType.mapping]: I think `VALUES` is the best. If not, it should at least have an underscore: `WOOD_TYPES`.

Comment (liach) [reply mappings/net/minecraft/util/WoodType.mapping]:
```suggestion
	FIELD field_21682 VALUES Ljava/util/Set;
```

**Review by Juuxel — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]
Nice work, looks good. I could have done with this the other day :D

**Pull request merged**